js/util/extractProps.ts

```diff
@@ -22,11 +29,18 @@ export const extractProps = (
   component: Component,
   config: PropExtractorConfig
 ): string => {
-  if (!component || !config[elementName]) {
+  if (!component) {
     return '';
   }
 
-  const filteredProps = pick(component.props, config[elementName].include);
+  let classCriteria: PropExtractorCriteria = { include: [] };
+  if (component.heapOptions) {
```

Comment:
Since `component.heapOptions` is of type `{ eventProps? }`, this should also assert that `component.heapOptions.eventProps` exists, otherwise `classCriteria` could be `undefined` instead of `{ include: [] }`.

Reply:
done - good catch.  I added a test to cover that case as well.